### PR TITLE
nali 0.4.2 (new formula)

### PR DESCRIPTION
**Pull request by kidonng**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

Add [nali](https://github.com/zu1k/nali), a tool for querying IP geographic information and CDN provider.

### CODE REVIEW

File: Formula/nali.rb [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+  def install
+    system "go", "build", *std_go_args(ldflags: "-s -w")
+    (bash_completion/"nali").write `#{bin}/nali completion bash`
+    (fish_completion/"nali.fish").write `#{bin}/nali completion fish`
+    (zsh_completion/"_nali").write `#{bin}/nali completion zsh`
+  end
+
+  def caveats
+    <<~EOS
+      Config will be written to the ~/.nali folder.
```
Comment (dawidd6) [inline Formula/nali.rb:20]: Caveats are for Homebrew-specific info. This is not.

File: Formula/nali.rb [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+    EOS
+  end
+
+  test do
+    ip = "1.1.1.1"
+    assert_match "#{ip} [美国 APNIC&CloudFlare公共DNS服务器]", shell_output("#{bin}/nali #{ip}")
```
Comment (dawidd6) [inline Formula/nali.rb:26]: Is there any output in English?

File: Formula/nali.rb [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+  url "https://github.com/zu1k/nali/archive/v0.4.2.tar.gz"
+  sha256 "25842823d61b1c05d8e261c28a2f24232838a1397cbe3b227a2c6288ec451fd6"
+  license "MIT"
+  head "https://github.com/zu1k/nali.git", branch: "master"
+
+  depends_on "go" => :build
+
+  def install
+    system "go", "build", *std_go_args(ldflags: "-s -w")
+    (bash_completion/"nali").write `#{bin}/nali completion bash`
```
Comment (dawidd6) [inline Formula/nali.rb:13]:
```suggestion
    (bash_completion/"nali").write Utils.safe_popen_read(bin/"nali", "completion", "bash")
```

Same below.

**Review by dawidd6 — CHANGES_REQUESTED** [review]

File: Formula/nali.rb (hunk 2 above)
Comment (kidonng) [reply Formula/nali.rb:20]: Some part of the program is in English, like the help text. The actual info text comes from external database, and [the only English database (GeoIP2) supported](https://github.com/zu1k/nali/blob/4cccb663b47ac93c6b37d92e5237d46b227a6234/README_en.md#multilingual-support) requires an account to download from MaxMind website. I guess this does make the program less useful for non-Chinese speaking users out of the box though.

It reads "US **APNIC&CloudFlare** Public **DNS** Server", bold text are in English so it's not like completely unrecognizable.

Comment (dawidd6) [reply Formula/nali.rb:20]: Yeah a comment would be nice.

**Review by dawidd6 — APPROVED** [review]

**Review by chenrui333 — APPROVED** [review]

**Comment by chenrui333** [comment]
@kidonng, thanks for your contribution to Homebrew! :tada: 🥇

Without awesome contributors like you, it would be impossible to maintain Homebrew to the high level of quality users have come to expect. Thank you!!!!

**Pull request closed without merging**